src/agent/scm/lib/svnwrapper.ts

```diff
@@ -0,0 +1,475 @@
+        var deferred: Q.Deferred<string[]> = Q.defer<string[]>();
+        
+        if (shell.test("-d", path.join(rootPath, administrativeDirectoryName))) {
+            // The rootPath contains .svn subfolder and we treat it as
+            // a working copy candidate.  
+            deferred.resolve([rootPath]);
+        }
+        else {
+            // Browse direct subfolder children of the rootPath
+            utilm.readDirectory(rootPath, false, true, utilm.SearchOption.TopDirectoryOnly)
+            .then((subFolders: string[]) => {
+
+                // The first element in the collection returned by the method is the rootPath, 
+                // which we've already tested. Ignore it.
+                subFolders.shift();
+
+                var count = subFolders.length;
+                if (count > 0) {
+                    subFolders.forEach((subFolder) => {
+                        if (shell.test("-d", path.join(subFolder, administrativeDirectoryName))) {
+                            // The subfolder contains .svn directory and we treat it as
+                            // a working copy candidate.
+                            candidates.push(subFolder); 
+                            if (--count == 0) {
+                                deferred.resolve(candidates);
+                            } 
+                        }
+                        else {
+                            // Merge working directory paths found in the subfolder into the common candidates collection.
+                            this._getSvnWorkingCopyPaths(subFolder)
+                            .then((moreCandidates) => {
+                                candidates = candidates.concat(moreCandidates);
+                                if (--count == 0) {
+                                    deferred.resolve(candidates);
+                                } 
+                            })
+                        }
+                        
+                    });
+                }
+                else {
+                    deferred.resolve(candidates);
+                }
+            });
+        }
+        
+        return deferred.promise;
+    }
+
+    private _getTargetUrl(folder: string): Q.Promise<string> {
+        if (!shell.test("-d", folder)) {
+            throw new Error("Folder " + folder + " does not exists");
+        }
+        
+        var deferred = Q.defer<string>();
+        
+        this._shellExec('info', [folder, "--depth", "empty", "--xml"])
+        .then((ret) => {
+            if (!this.isSuccess(ret)) {
+                deferred.resolve(null);
+            }
+            else if (ret.output) {
+                xmlrReader.read(ret.output, (err, res) => {
+                    if (err) {
+                        deferred.reject(err);
+                    }
+                    else {
+                        try {
+                            return deferred.resolve(res.info.entry.url.text());
```

Comment:
Is there a possibility that the XML won't be in the format we expect? That would result in stuff like "could not find property entry of undefined".  Maybe we should check for this...